pypdf/_xobj_image_helpers.py

```diff
@@ -292,10 +290,17 @@ def _handle_jpx(
         mode = "RGBA"
     # we need to convert to the good mode
     try:
-        if img1.mode != mode:
+        if (img1.mode == mode) or (img1.mode in ("L", "P") and mode in ("L", "P")):
```

Comment:
DRY nitpick but this should probably made into a constant at the top of the file... something like 
```python
BAD_CONVERSION_MODES = ("L", "P")
```

Reply:
They only apply to JPX, thus this probably would have to be part of the constants as well.

Reply:
These are not bad modes, just some indexed/palettes based modes. OK to put a constant name and suggested by @stefan6419846, next to the function